### PR TITLE
Fixes #10904

### LINKED ISSUE #10904: Issue NameMice can unbuckle people
(WEB REPORT BY: 9600bauds REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> Issue NameMice can unbuckle people
> 
> General description of the issue
> Issue NameMice can unbuckle people
> 
> What you expected to happen
> Issue NameMice can unbuckle people
> 
> What actually happened
> Issue NameMice can unbuckle people
> 
> Steps to reproduce if possible
> Issue NameMice can unbuckle peopleIssue NameMice can unbuckle peopleIssue NameMice can unbuckle people

### PR DESCRIPTION
**Pull request by 9600bauds**
Fixes #10904 Pull requests must be atomic.  Change one set of related things at a time.  Bundling sucks for everyone.
This means, primarily, that you shouldn't fix bugs and add content in the same PR. When we mean 'bundling', we mean making one PR for multiple, unrelated changes.

Test your changes.  PRs that do not compile will not be accepted.
Testing your changes locally is incredibly important. If you break the serb we will be very upset with you.

Large changes require discussion.  If you're doing a large, game-changing modification, or a new layout for something, discussion with the community is required as of 26/6/2014.  Map and sprite changes require pictures of before and after.  MAINTAINERS ARE NOT IMMUNE TO THIS.  GET YOUR ASS IN IRC.

Merging your own PRs is considered bad practice, as it generally means you bypass peer review, which is a core part of how we develop.

It is also suggested that you hop into irc.rizon.net #vgstation to discuss your changes, or if you need help.

### CODE REVIEW

**Comment by ghost** [comment]
You know you can blank the default form if you know what you are doing, right ?

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 1]
```diff
@@ -55,6 +55,10 @@
 	if(!locked_atoms.len)
 		return
 
+	if(user.size <= SIZE_TINY)
+		to_chat(user, "<span class='warning'>You much too small to do that.</span>")
```
Comment (ghost) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: I speak English very good, should be "You are too small to do that."

**Comment by PJB3005** [comment]
He's memeing you.

**Comment by PJB3005** [comment]
Bauds please your intellect has dropped badly recently.

Here have some alkysine.

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm (hunk 1 above)
Comment (Intigracy) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Sheeeeet, you is not big enough to do that dawg.

**Comment by clusterfack** [comment]
Why did it break

**Comment by 9600bauds** [comment]
I broke it

**Comment by Intigracy** [comment]
bauds a borker

**Comment by 9600bauds** [comment]
borker borker

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 2]
```diff
@@ -55,6 +55,10 @@
 	if(!locked_atoms.len)
 		return
 
+	if(user.size <= SIZE_TINY)
+		to_chat(user, "<span class='warning'>You are much too small to do that.</span>")
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Honestly the much just seems redundant.

**Comment by JustSumBody** [comment]
> controversial 
> discussion

You what

**Comment by ComicIronic** [comment]
Should you not also fix the buckling in at the same time?

**Comment by Intigracy** [comment]
Can they buckle people?

The issue was about unbuckling.

**Comment by ComicIronic** [comment]
I don't know, can they? There's nothing immediate in the code that suggests they can't.

**Comment by 9600bauds** [comment]
what is returt??

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 3]
```diff
@@ -72,21 +76,29 @@
...
 
 /obj/structure/bed/proc/buckle_mob(mob/M as mob, mob/user as mob)
-	if(!ismob(M) || !Adjacent(user) || (M.loc != src.loc) || user.restrained() || user.lying || user.stat || M.locked_to || istype(user, /mob/living/silicon/pai) )
+	if(!Adjacent(user) || user.incapacitated() || istype(user, /mob/living/silicon/pai))
+		return
+
+	if(!ismob(M) || (M.loc != src.loc)  || M.locked_to)
+		return
+
+	if(locked_atoms.len)
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Shouldn't be checking it like this, use the locking category to see it.

Comment (9600bauds) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Explain further please, I'm not too familiar with locking categories

Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Use `get_locked(lock_type)`

Comment (9600bauds) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: There doesn't seem to be any locking categories for chairs or beds

Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/game/objects/structures/stool_bed_chair_nest/bed.dm#L172

https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/game/objects/structures/stool_bed_chair_nest/chairs.dm#L229

Sure?

Comment (9600bauds) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Whoops, they're just not used anywhere yet, my bad

Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: > Whoops, they're just not used anywhere yet

Please what are you on about.

Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/game/objects/structures/stool_bed_chair_nest/bed.dm#L104

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 4]
```diff
@@ -72,21 +76,29 @@
 	add_fingerprint(user)
 
 /obj/structure/bed/proc/buckle_mob(mob/M as mob, mob/user as mob)
-	if(!ismob(M) || !Adjacent(user) || (M.loc != src.loc) || user.restrained() || user.lying || user.stat || M.locked_to || istype(user, /mob/living/silicon/pai) )
+	if(!Adjacent(user) || user.incapacitated() || istype(user, /mob/living/silicon/pai))
+		return
+
+	if(!ismob(M) || (M.loc != src.loc)  || M.locked_to)
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:82]: `src.`

Comment (9600bauds) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:82]: meme

**Pull request merged**